pkgs/servers/draupnir/default.nix

```
@@ -0,0 +1,98 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, yarn
+, matrix-sdk-crypto-nodejs
+, mkYarnPackage
+, fetchYarnDeps
+}:
+
+# docs: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#yarn2nix-javascript-yarn2nix
+
+mkYarnPackage rec {
+  pname = "draupnir";
+  version = "1.85.1";
+  # NOTE: Hard fork of Mjolnir.
+  src = fetchFromGitHub {
+    owner = "the-draupnir-project";
+    repo = "Draupnir";
+    rev = "v${version}";
+    sha256 = "sha256-cXTkdvFYLsfoeVk5WAIABqw+oFynaMlBeBocXaJUO6A=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    makeWrapper
+    yarn
+    matrix-sdk-crypto-nodejs
+  ];
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    # nix hash file yarn.lock
+    hash = "sha256-j4xiEP54b7nn+bKkTUemwFNpA7s8DlsYnYX4KZgUhHU=";
+  };
+  yarnLock = src + "/yarn.lock";
+  packageJSON = src + "/package.json";
```

```suggestion
  packageJSON = ./package.json;
```

Can you try importing the `package.json` into this repo?

---

Added the package.json to nixpkgs, not sure if eg. yarn.lock should be added too?

---

Seems like importing package.json was sufficient. The lockfile would be a different story, since it is quite large.